frontend/js/src/explore/artist-similarity/SimilarArtistsGraph.tsx

```diff
@@ -0,0 +1,164 @@
+        }
+      )}
+      onClick={onClick ? (event) => onClick(node, event) : undefined}
+      onMouseEnter={
+        onMouseEnter ? (event) => onMouseEnter(node, event) : undefined
+      }
+      onMouseMove={
+        onMouseMove ? (event) => onMouseMove(node, event) : undefined
+      }
+      onMouseLeave={
+        onMouseLeave ? (event) => onMouseLeave(node, event) : undefined
+      }
+      width={to([animatedProps.size], (size) => size / 2)}
+      height={to([animatedProps.size], (size) => size / 2)}
+    >
+      <animated.circle
+        data-testid={`node.${node.id}`}
+        r={to([animatedProps.size], (size) => size / 2)}
+        fill={animatedProps.color}
+        strokeWidth={animatedProps.borderWidth}
+        stroke={animatedProps.borderColor}
+        opacity={animatedProps.opacity}
+      />
+      <animated.text
+        textAnchor="middle"
+        dominantBaseline="central"
+        fontSize={to([animatedProps.size], (size) => size / 6)}
+        style={{ pointerEvents: "none", fill: "white" }}
+      >
+        {// SVG text does not allow for easy multi line text
+        // So, we split the text into words and render each word as a separate tspan element in a different line
+        // We use ellipsis if word length exceeds MAX_WORD_LENGTH or if number of words(each in a separate line) exceed MAX_LINES
+        node.data.artist_name.split(" ").map((word, index) =>
+          index < MAX_LINES ? (
+            <animated.tspan
+              x={0}
+              dy={
+                index === 0 ? "0" : to([animatedProps.size], (size) => size / 6)
+              }
+            >
+              {word.length > MAX_WORD_LENGTH
+                ? `${word.substring(0, MAX_WORD_LENGTH - 3)}...`
+                : word}
+            </animated.tspan>
+          ) : (
+            index === MAX_LINES && (
+              <animated.tspan
+                x={0}
+                dy={to([animatedProps.size], (size) => size / 6)}
+              >
+                ...
+              </animated.tspan>
+            )
+          )
+        )}
+      </animated.text>
+    </animated.g>
+  );
+}
+
+function CustomNodeTooltipComponent({ node }: NodeTooltipProps<NodeType>) {
+  return (
+    <div
+      style={{
+        background: node.color,
+        color: "#ffffff",
+        padding: "9px 12px",
+        borderRadius: "3px",
+      }}
+    >
+      <strong>{node.data.artist_name}</strong>
+      <br />
+      {isFinite(node.data.score) && <>Score: {node.data.score}</>}
+    </div>
+  );
+}
+
+function SimilarArtistsGraph({
+  data,
+  onArtistChange,
+  background,
+  graphParentElementRef,
+}: GraphProps) {
+  const chartProperties: OmitHeightWidth<NetworkSvgProps<
+    NodeType,
+    LinkType
+  >> = {
+    data,
+    repulsivity: 350,
+    iterations: 120,
+    centeringStrength: 0.11,
+    nodeBorderWidth: 5,
+    linkThickness: 2,
+    distanceMin: 20,
+    distanceMax: 550,
```

Comment:
Both the toolbar at the top and the graph underneath are looking a bit rough on smaller screens:
<img width="301" alt="image" src="https://github.com/metabrainz/listenbrainz-server/assets/6179856/9c2658d7-df6d-4995-9227-ca05d7476aa0">

I wonder if the values for repulsivity, distanceMax and nodeSize should be different depending on the available screen size, or maybe the size of the nodes, or…
I tried looking for some examples or to see if there are other solutions, but didn't find anything useful.

I do notice that I get some cropped nodes also in other non-mobile screen sizes:

<img width="789" alt="image" src="https://github.com/metabrainz/listenbrainz-server/assets/6179856/3dfae830-2879-4481-af14-0e3d1cfb5c6b">

I wonder if it has to do with the size of the nodes being ignored, meaning the center of the top-most node is just about in frame but the node scaled from that center point is out of bounds